jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/PerProcessInstanceRuntimeManager.java

```diff
@@ -182,18 +182,23 @@ public void signalEvent(String type, Object event) {
         }
     
         // next find out all instances waiting for given event type
+        runtimeEngine = null;
         List<String> processInstances = ((InternalMapper) mapper).findContextIdForEvent(type, getIdentifier());
         for (String piId : processInstances) {
-            runtimeEngine = getRuntimeEngine(ProcessInstanceIdContext.get(Long.parseLong(piId)));
             try {
+                runtimeEngine = (RuntimeEngineImpl) getRuntimeEngine(ProcessInstanceIdContext.get(Long.parseLong(piId)));
                 // signal execution can rise errors
                 if (!signalledEngines.contains(runtimeEngine)) {
-                    runtimeEngine.getKieSession().signalEvent(type, event);
                     signalledEngines.add(runtimeEngine);
+                    runtimeEngine.getKieSession().signalEvent(type, event);
                 }
+            } catch (org.drools.persistence.api.SessionNotFoundException ex) {
```

Comment:
This needs to catch `org.kie.internal.runtime.manager.SessionNotFoundException` in my tests:
```
ERROR [org.kie.server.services.jbpm.kafka.KafkaServerConsumer] (pool-18-thread-10) Exception deserializing event: org.kie.internal.runtime.manager.SessionNotFoundException: No session found for context 143
        at org.jbpm.runtime.manager.impl.PerProcessInstanceRuntimeManager$PerProcessInstanceInitializer.initKieSession(PerProcessInstanceRuntimeManager.java:584)
        at org.jbpm.runtime.manager.impl.RuntimeEngineImpl.internalGetKieSession(RuntimeEngineImpl.java:169)
        at org.jbpm.runtime.manager.impl.RuntimeEngineImpl.getKieSession(RuntimeEngineImpl.java:76)
        at org.jbpm.runtime.manager.impl.PerProcessInstanceRuntimeManager.signalEvent(PerProcessInstanceRuntimeManager.java:186)
        at org.jbpm.kie.services.impl.ProcessServiceImpl.signalEvent(ProcessServiceImpl.java:383)

```

Reply:
added that exception